src/main/scala/net/snowflake/spark/snowflake/SnowflakeWriter.scala

```diff
@@ -288,10 +288,52 @@ private[snowflake] class SnowflakeWriter(
   /**
     * Generate the COPY SQL command
     */
-  private def copySql(sqlContext: SQLContext,
+  private def copySql(data: DataFrame,
+                      saveMode: SaveMode,
                       params: MergedParameters,
                       filesToCopy: (String, String),
                       tempStage: Option[String]): String = {
+
+    if (saveMode != SaveMode.Append && params.columnMap.isDefined)
+      throw new UnsupportedOperationException("The column mapping only works in append mode.")
+
+    def getMappingToString(list: Option[List[(Int, String)]]): String = {
+      if (list.isEmpty || list.get.isEmpty) ""
+      else {
+        val map = list.get
+        val buffer: mutable.StringBuilder = new mutable.StringBuilder()
+        buffer.append("(")
+        buffer.append(map.head._2)
+        (1 until map.size).foreach(i => {
+          buffer.append(", ")
+          buffer.append(map(i)._2)
```

Comment:
Let's ensure proper escaping of these identifiers. See: quotedName() and its usage in SnowflakeJDBCWrapper.

Reply:
done